src/Type/Type.php

```diff
@@ -91,6 +93,11 @@ public function toArray(): Type;
 	 */
 	public function inferTemplateTypes(Type $receivedType): TemplateTypeMap;
 
+	/**
+	 * @return TypeArgumentReference[]|iterable
+	 */
+	public function getReferencedTypeArguments(TemplateTypeVariance $positionVariance): iterable;
```

Comment:
Can you explain what this method does? Also, is there any advantage in `iterable` over `array`? Other methods use arrays mainly...

Reply:
I found `yield` very handy in this case, as I didn't had to array_merge everything. I'm rewriting to use `arrays` instead, for consistency.

Reply:
Also changed the name and added some doc